app/src/main/java/com/example/drawingactivity/MainActivity.kt

```diff
@@ -17,11 +25,63 @@ class MainActivity : AppCompatActivity() {
         viewModel.currentBitmap?.let { drawableArea.setBitmap(it) }
         val colorSelector = findViewById<PrimaryColorCircleView>(R.id.colorCircle)
         colorSelector.onChangeColor = { viewModel.cyclePrimaryColor() }
+        findViewById<Button>(R.id.add_layer).setOnClickListener {
+            colorPickerDialog()
+        }
+
+        if (viewModel.isDialogOpen) { colorPickerDialog() }
 
         viewModel.onUpdate = {
             colorSelector.circlePrimaryColor = it.primaryColor
             drawableArea.pathPaint.color = it.primaryColor
             drawableArea.userDrawnBitmap = it.currentBitmap
         }
     }
+
+
+    @SuppressLint("ClickableViewAccessibility")
+    private fun colorPickerDialog() {
+        viewModel.isDialogOpen = true
+        val inflater = LayoutInflater.from(this@MainActivity)
+        val view = inflater.inflate(R.layout.add_layer_dialog, null)
+        val dialog = AlertDialog.Builder(this).apply {
+            setTitle("Add Layer")
```

Comment:
String resource

Reply:
Now uses a string resource for dialog title